### PR TITLE
Adding new negative cases for hawkular provider creation

### PR DESCRIPTION
**Pull request by FilipB**
{{pytest: cfme/tests/middleware/test_middleware_provider.py -v --use-provider hawkular }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 00e213ad5574307c0f4759ab33ff7db09c2e2789:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/middleware/test_middleware_provider.py [hunk 1]
```diff
@@ -10,11 +16,89 @@
...
+    cred = prov.Credential(
+        principal='bad',
+        secret=fauxfactory.gen_alphanumeric(5),
+        verify_secret=fauxfactory.gen_alphanumeric(6)
+    )
+    sel.force_navigate('{}_provider_new'.format(prov.page_name))
+    fill(prov.properties_form, {"type_select": "Hawkular"})
+    fill(prov.properties_form, {"hostname_text": "test"})
+    fill(prov.properties_form, {"name_text": "test"})
+    fill(cred.form, cred)
```
Comment (mfalesni) [inline cfme/tests/middleware/test_middleware_provider.py]: This is not how fill is intended to be used. You shall pass all the values to fill, it goes through the form and fills the values then.

**Comment by mfalesni** [comment]
The commit is not signed and the PR description does not have the proper metadata for testing since we don't use middleware as one of our "default" providers. You shall put something like this in your PR's description:

{{pytest: cfme/tests/middleware/test_middleware_provider.py -v --use-provider hawkular }}

**Comment by dajoRH** [comment]
Lint report for commit dda2147334b14f1cd84d7393dc796d2528a570ef:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by FilipB** [comment]
Errors against upstream are caused by https://github.com/ManageIQ/integration_tests/issues/3405.
test_provider_add_with_bad_port[hawkular] fails because the blocking issue https://github.com/ManageIQ/manageiq/issues/8241 is closed but the fix is not yet available in appliance

**Comment by psav** [comment]
This should be fixed by #3430 or at least upstream should be fixed by this.

**Comment by dajoRH** [comment]
Lint report for commit dfc962ce50c288e9bd9a098151bb3772e3ccbf6e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5d9a72ea5e192d66ef0cf37d70ef0024fef5f650:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by FilipB** [comment]
Rebased  on master.
All tests are passing locally against 5.7.0.7-beta1.20161025153249_9376fbd

**Comment by dajoRH** [comment]
Lint report for commit e45f0f5b4af9e8ed25effa3e59a20daf6c4d7cbb:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by FilipB** [comment]
@psav @mfalesni 
May I ask for review please?
PRT is failing because the wait_for_ajax() is timing out on the spinner. But some runs passed e.g. http://10.16.4.32/trackerbot/pr/run/8241

**Pull request merged**